File: medium/1344-angle-between-hands-of-a-clock/solution.py
class Solution:
    def angleClock(self, hour: int, minutes: int) -> float:
        """
        1344. Angle Between Hands of a Clock
        Time: O(1)
        Space: O(1)
        """
        minute_angle = 6 * minutes
        hour_angle = 30 * (hour % 12) + 0.5 * minutes

        diff = abs(hour_angle - minute_angle)
        return min(diff, 360 - diff)

File: medium/1344-angle-between-hands-of-a-clock/solution.java
/**
 * 1344. Angle Between Hands of a Clock
 * Time: O(1)
 * Space: O(1)
 */
class Solution {
    public double angleClock(int hour, int minutes) {
        double minuteAngle = 6.0 * minutes;
        double hourAngle = 30.0 * (hour % 12) + 0.5 * minutes;

        double diff = Math.abs(hourAngle - minuteAngle);
        return Math.min(diff, 360.0 - diff);
    }
}
